decidim/decidim
# Setup elections

#### :tophat: What? Why?
This PR adds a GraphQL client that connects to the Bulletin Board.

#### :pushpin: Related Issues


#### Testing
- **Create at least 2 Trustees** for a Participatory space (for instance, for Processes)
- **Go to the Elections tab** in the participatory space selected.
- Click the new Setup button.

<img width="1375" alt="New setup election button" src="https://user-images.githubusercontent.com/60884395/96625273-413e3d80-130e-11eb-9914-3b4019932101.png">


- You will find a list of all the requirements needed for the Setup being done.
<img width="1375" alt="Election not valid" src="https://user-images.githubusercontent.com/60884395/96610991-83f71a00-12fc-11eb-9051-89b4b9ddfcea.png">

- If there's something that is pending, it will be shown with an X for that line.
- Requirements that are achieved will be shown with a check sign.
- When at least 1 requirement is not met, the Setup election button will be disabled.
- When all the requirements are met, the Setup button will be active.


Example of a Test of requirements not met:
- For an election, **make sure it has at least one question, but that question has just 1 answer or none.**
- When you go to the Setup button, **there must be an X for the checking of the number of answers and the Setup election button must be disabled.**

Example of a Test of being met all requirements:
- You can just follow the indications when clicking the setup button, if there's any error, solve it, and once all of them are solved, the setup button will be enabled.
<img width="1375" alt="Election valid" src="https://user-images.githubusercontent.com/60884395/96611004-86f20a80-12fc-11eb-858b-9730e836f14b.png">

- After clicking the Setup election button, if everything went well, you will see a message in green:

<img width="1375" alt="Setup Election Successful" src="https://user-images.githubusercontent.com/60884395/96624934-c2e19b80-130d-11eb-8149-e161c7b6c109.png">

- You can verify the created election going to the Bulletin Board GraphiQL Client, and doing the query for the new Election:

<img width="1375" alt="graphiql election checking" src="https://user-images.githubusercontent.com/60884395/96626442-caa23f80-130f-11eb-9bdf-122c708fca9a.png">


#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.


### :camera: Screenshots
*Please add screenshots of the changes you're proposing*



:hearts: Thank you!
